pkg/target/target_flags.go

```diff
@@ -115,6 +124,8 @@ func (tf *targetFlagsImpl) OverrideTarget(current Target) (Target, error) {
 		if err := current.Validate(); err != nil {
 			return nil, fmt.Errorf("invalid target flags: %w", err)
 		}
+
+		current = current.WithControlPlane(tf.controlPlane)
```

Comment:
Why is this not done before `current.Validate`?
Why do we always override the `control-plane` even if not specified in target flags?

Reply:
You are right, this is a bug introduced recently while I was experimenting with the target flags. Fixed it.